[docs] add submodule in engine.rst

ydcjeff opened the pull request:
Fixes #1459 

Description: Add submodules between classes in engine.rst

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

vfdev-5 commented on docs/source/engine.rst:
```diff
@@ -3,18 +3,24 @@ ignite.engine
+ignite.engine.deterministic
+
 .. currentmodule:: ignite.engine.deterministic
 
```
(could not comment on unmodified part)
```diff
- and helper methods:
+ and helper methods to define supervised trainer and evaluator:

vfdev-5 commented on docs/source/engine.rst:
```diff
@@ -3,18 +3,24 @@ ignite.engine
     :nosignatures:
     :autolist:
 
+ignite.engine.deterministic
```
```suggestion
ignite.engine.deterministic (helper methods for deterministic training)
```

vfdev-5 reviewed (commented):
PR looks good @ydcjeff ! I was just thinking to add few words explaining each submodule. I left an example in the comments.

ydcjeff commented on docs/source/_templates/_static/css/ignite_theme.css:
```diff
@@ -131,8 +135,11 @@ article.pytorch-article table.longtable.docutils.align-default colgroup {
+  table.longtable.docutils.align-default
+  tbody
+  td:first-child {
+  width: 45%;
```
I only edit this `width`, the rest are formatted by pre-commit

ydcjeff commented:
@vfdev-5 I found we can also add in contrib.engine and I added it. Also the table with width 30% doesn't look good in contrib.engine so I changed to 45%, the rest are formatted by pre-commit.

vfdev-5 commented:
@ydcjeff thanks for the updates ! When used 45% for tables, other tables start looking a bit different:
https://deploy-preview-1464--pytorch-ignite-preview.netlify.app/engine.html
vs 
https://pytorch.org/ignite/master/engine.html
Maybe,  a smaller value like 40% can be OK. @trsvchn what do you think ?

ydcjeff commented:
@vfdev-5 yes that wide all applies to all `.. currentmodule`, maybe 40 looks good

trsvchn commented:
@vfdev-5 Yes 45% is too much let's try 40%. Another option to specifically change that table using css selectors (if possible, I need to check this option)

vfdev-5 reviewed (approved):
Thanks !
